### PR TITLE
Calculate WebNN padding values rather than using autoPad

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@huningxin, PTAL, thanks!

This is referenced from XNNPack delegate: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc#L1206.

**Review by huningxin — APPROVED** [review]
lgtm, thanks!

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc [hunk 1]
```diff
@@ -399,6 +400,75 @@ class Subgraph {
     }
   }
 
+  // Calculate WebNN padding values specific for kTfLitePaddingSame.
+  static TfLiteStatus CalculateWebnnPaddings(
```
Comment (fdwr) [inline tensorflow/lite/delegates/webnn/webnn_delegate.cc:404]: Oh, so I guess TF *didn't* already have a helpful existing function to compute this :(.

Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc:404]: They do have the helper function, it's [`ComputePaddingHeightWidth`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/padding.h#L62).

`CalculateWebnnPaddings`: this function is used for handling the difference between `TransposeConv` and `Conv`, as `TransposeConv` uses `output_height` and `output_width` to calculate expected input height and input width.

Comment (fdwr) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc:404]: I see. I just saw this sizeable function `CalculateWebnnPaddings` and thought you had to write a bunch of new logic yourself, but now I see that *most* of it here is actually error checking.

**Pull request merged**